### PR TITLE
[Typo] Replace PR#514-525 with one large PR

### PR DESCRIPTION
**Pull request by practicalswift**
As requested in #526.

### CODE REVIEW

**Comment by practicalswift** [comment]
@gottesmm The code is verified passing `utils/build-script -RT` and the changes suggested above have been made. Please let me know if any further adjustments are needed :-)

**Comment by gottesmm** [comment]
@benlangmuir I just looked at the first commit.

**Comment by gottesmm** [comment]
(i.e. the rest of the commits need to be looked at).

**Comment by practicalswift** [comment]
Squash into a single commit.

**Comment by benlangmuir** [comment]
@gottesmm I looked at the rest of the changes, which LGTM.  My question was about a specific change that you requested (in #501):

>This should probably also have its grammar fixed. I would suggest:
>
>// We work around the canonicalization of PolymorphicFunctionType
>// by generating a GenericSignature and transforming the input, output
>// types.

Do we really want to drop the "FIXME" in this comment?

**Comment by gottesmm** [comment]
That was not my intention. I just meant the text after the FIXME.

File: lib/Parse/ParseSIL.cpp [hunk 1]
```diff
@@ -1516,8 +1516,8 @@ bool getApplySubstitutionsFromParsed(
   return false;
 }
 
-// FIXME: we work around canonicalization of PolymorphicFunctionType
-// by generating GenericSignature and transforming the input, output
+// We work around the canonicalization of PolymorphicFunctionType
+// by generating a GenericSignature and transforming the input, output
```
Comment (benlangmuir) [inline lib/Parse/ParseSIL.cpp]: @practicalswift could you add the FIXME back here (per @gottesmm's clarification), and then re-squash?

**Comment by practicalswift** [comment]
@benlangmuir Re-added FIXME as per @gottesmm and re-squashed. Please let me know if any further changes are needed :-)

**Pull request merged**